src/components/Admin/Admin.tsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from "react";
+import Card from "../UI/Card/Card";
+import { authHeader } from "../../helpers/auth";
+import { useNavigate } from "react-router-dom";
+import AllUsers from "./AllUsers";
+import classes from './Admin.module.css'
+import UserDetails from "./UserDetails";
+import { User } from "../../types/User";
+import axios from "axios";
+
+const Admin = () => {
+    const nav = useNavigate()
+    const [chosenUser, setChosenUser] = useState<User>({
+        _id: '',
+        active: false,
+        password: '',
+        phone: '',
+        role: 'hr',
+        username: ''
+    });
+    const [userData, setUserData] = useState<User[]>([]);
+    const [updated, setUpdated] = useState<boolean>(false);
+    const [mode, setMode] = useState("view");
+
+    useEffect(() => {
+        const hdrs = authHeader()
```

Comment:
Need to check if the user is admin or not as well. Only admins should be able to enter this page, and only admins should have the "admin dashboard" link at the navbar